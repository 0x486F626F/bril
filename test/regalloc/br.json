{
  "functions": [
    {
      "instrs": [
        {
          "dest": "va",
          "op": "const",
          "type": "int",
          "value": 4
        },
        {
          "dest": "vb",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "dest": "b",
          "op": "const",
          "type": "bool",
          "value": true
        },
        {
          "args": [
            "b",
            "right",
            "left"
          ],
          "op": "br"
        },
        {
          "label": "left"
        },
        {
          "args": [
            "va"
          ],
          "dest": "vd",
          "op": "id",
          "type": "int"
        },
        {
          "args": [
            "end"
          ],
          "op": "jmp"
        },
        {
          "label": "right"
        },
        {
          "args": [
            "vb"
          ],
          "dest": "vd",
          "op": "id",
          "type": "int"
        },
        {
          "label": "end"
        },
        {
          "args": [
            "vd"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
