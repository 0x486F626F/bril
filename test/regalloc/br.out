main {
	__r02: int = const 4;
	__r03: int = const 5;
	__r01: bool = const true;
	br __r01 right left;
left:
	b: bool = id __r01;
	__r01: int = id __r02;
	jmp end;
right:
	__r01: int = id __r03;
end:
	print __r01;
}
